src/mesh/boundary_standard.cxx

```diff
@@ -248,10 +260,13 @@ void BoundaryOpWithApply<Derived, needs_delta>::applyTemplate(T &f, BoutReal t)
           delta = 0.;
         }
 
-        for(int z=0; z<f.getNz(); z++) {
-          BoutReal znorm = localmesh->GlobalZ(z);
+        for(int z=0; z<nz; z++) {
           if (fg) {
+            BoutReal znorm = BoutReal(z)/BoutReal(nz);
+#pragma GCC diagnostic push
```

Comment:
I think these would need to be removed should this go into the main code -- we really want to avoid this level of (compiler family specific) detail where possible. I think the simplest fix here would be to just explicitly say xnorm=ynorm=0 in the initial declaration of the variables.

Reply:
Yes, absolutely agree. It's good practice to always initialise variables

Reply:
:+1: sorry, I think I must have been in caffeine withdrawal when I thought that was a good idea :tired_face: